### ISSUE TITLE
DNS results returned in incorrect order

### ISSUE THREAD

**Issue by mithrandi** [rancher/rancher#8946]
**Rancher versions:**
rancher/server: 1.6.0
rancher/agent: 1.2.2

**Infrastructure Stack versions:**
healthcheck: 0.3.0
ipsec: 0.1.0
network-services: 0.2.0
scheduler: 0.5.1
kubernetes (if applicable): not used

I'm running Rancher 1.6.0 and having a weird DNS issue in containers started by Rancher. Specifically, it appears DNS lookups involving CNAMEs are sometimes returning the records in the wrong order:

```
root@b5decf3079c1:~# host s3.amazonaws.com
s3.amazonaws.com is an alias for s3-1.amazonaws.com.
s3-1.amazonaws.com has address 54.231.120.26
root@b5decf3079c1:~# host s3.amazonaws.com
s3-1.amazonaws.com has address 54.231.120.26
s3.amazonaws.com is an alias for s3-1.amazonaws.com.
```

The two orders seem to come up randomly. I wouldn't have expected this to be a problem, but it seems the ordering where the CNAME is first causes many application lookups to fail completely:

```
root@b5decf3079c1:~# ping -c 1 s3.amazonaws.com
PING s3.amazonaws.com (52.216.81.75): 56 data bytes
64 bytes from 52.216.81.75: icmp_seq=0 ttl=46 time=260.373 ms
--- s3.amazonaws.com ping statistics ---
1 packets transmitted, 1 packets received, 0% packet loss
round-trip min/avg/max/stddev = 260.373/260.373/260.373/0.000 ms
root@b5decf3079c1:~# ping -c 1 s3.amazonaws.com
ping: unknown host
```

If I test this on the host, or in a container running the same image but not using the Rancher CNI, then the CNAME record always comes first, and the lookups always succeed.

I suspect this may be caused by the fix for #7990 randomly reordering _all_ of the returned records , not just the `A` and `AAAA` records.

**Comment by mithrandi** [comment]
Oops, sorry. This is a duplicate of #8840 and appears to be fixed in rancher/rancher-dns#63.

**Comment by mithrandi** [comment]
I upgraded to rancher/dns:0.15.1 and the problem seems to be resolved, so I guess the fix worked.
